fix the warnings in the codebase

![Image](https://user-images.githubusercontent.com/106985561/202490898-ff95af41-7e86-4380-ba40-72a8edc4d0cf.png)

There are some warnings to fix in our codebase, when we run the command `yarn lint:test`

3 warnings remain:

```
/Users/iandilling/Workspace/ui-kit-2022/packages/ui-kit-components/src/components/Graph/graph.tsx
  246:36  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/Users/iandilling/Workspace/ui-kit-2022/packages/ui-kit-theme/src/index.ts
  18:35  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  25:34  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

✖ 3 problems (0 errors, 3 warnings)
```

--- a/packages/ui-kit-components/src/components/Graph/graph.tsx
+++ b/packages/ui-kit-components/src/components/Graph/graph.tsx
@@ -6,6 +6,7 @@ import {
   LinearScale,
   LineElement,
   PointElement,
+  ScriptableScaleContext,
   Title,
   Tooltip,
 } from 'chart.js';
@@ -120,13 +121,6 @@ export interface GraphProps {
   previousData?: number;
 }
 
-interface GridColorCtx {
-  index: number;
-  tick: {
-    label: string;
-  };
-}
-
 /**
  *
  * @param yLabelStep represents the step size between the values of labels in y-axis.
@@ -170,7 +164,7 @@ const Graph: React.FC<GraphProps> = ({
           autoSkip: true,
         },
         grid: {
-          color: (ctx: GridColorCtx) => {
+          color: (ctx: ScriptableScaleContext) => {
             if (ctx.tick.label === '') {
               return 'rgba(0, 0, 0, 0)';
             } else {
@@ -190,7 +184,7 @@ const Graph: React.FC<GraphProps> = ({
         },
         grid: {
           drawOnChartArea: true,
-          tickColor: (ctx: GridColorCtx) => {
+          tickColor: (ctx: ScriptableScaleContext) => {
             if (ctx.tick.label === '') {
               return 'rgba(0, 0, 0, 0)';
             } else {
@@ -199,7 +193,7 @@ const Graph: React.FC<GraphProps> = ({
                 : alpha((palette as Palette & PaperOption).paper.white, 0.2);
             }
           },
-          color: (ctx: GridColorCtx) => {
+          color: (ctx: ScriptableScaleContext) => {
             if (ctx.index === 0) {
               return palette.mode === 'light'
                 ? alpha((palette as Palette & PaperOption).paper.black, 0.2)
@@ -220,7 +214,7 @@ const Graph: React.FC<GraphProps> = ({
       legend: {
         display: false,
         labels: {
-          fontColor:
+          color:
             palette.mode === 'light'
               ? alpha(palette.common.black, 0.55)
               : alpha(palette.common.white, 0.55),
@@ -259,7 +253,7 @@ const Graph: React.FC<GraphProps> = ({
   };
 
   const datasets = getData(labelsConverted, data, palette);
-  return <Line options={options as any} data={datasets} />;
+  return <Line options={options} data={datasets} />;
 };
 
 export default Graph;

--- a/packages/ui-kit-theme/src/index.ts
+++ b/packages/ui-kit-theme/src/index.ts
@@ -15,14 +15,14 @@ export * from './constants';
 export { palette, spacing };
 
 export const light = createTheme({
-  components: LIGHT_COMPONENTS as any,
+  components: LIGHT_COMPONENTS,
   palette: palette.light,
   typography,
   spacing,
 });
 
 export const dark = createTheme({
-  components: DARK_COMPONENTS as any,
+  components: DARK_COMPONENTS,
   palette: palette.dark,
   typography,
   spacing,

--- a/packages/ui-kit-theme/src/overrides/MuiChip.ts
+++ b/packages/ui-kit-theme/src/overrides/MuiChip.ts
@@ -6,12 +6,11 @@ import {
   Theme,
 } from '@mui/material';
 
+type PaletteColorOption = keyof PaletteColorOptions;
+
 type OverrideContext = {
   theme: Theme;
-  ownerState: {
-    variant?: ChipProps['variant'];
-    color: PaletteColorOptions[keyof PaletteColorOptions];
-  };
+  ownerState: ChipProps;
 };
 
 export default {
@@ -32,7 +31,7 @@ export default {
         // Border color transitions to the opposite color variation from the theme on active and hover
         ...(ownerState.color !== 'default' && {
           borderColor:
-            theme.palette[ownerState.color][
+            theme.palette[ownerState.color as PaletteColorOption][
               theme.palette.mode === 'light' ? 'dark' : 'light'
             ],
         }),
@@ -50,7 +49,7 @@ export default {
       fontSize: theme.typography.button.fontSize,
       fontFamily: theme.typography.button.fontFamily,
       fontWeight: theme.typography.button.fontWeight,
-      textTransform: 'uppercase',
+      textTransform: 'uppercase' as const,
 
       lineHeight: '13px',
       padding: '4px',
@@ -65,23 +64,30 @@ export default {
                 // Color changes to white on hover in light mode
                 color: alpha(theme.palette.common.white, 0.95),
                 // Background color darkens on hover in light mode
-                backgroundColor: (theme.palette[ownerState.color] as PaletteColor).dark,
+                backgroundColor: (
+                  theme.palette[ownerState.color as PaletteColorOption] as PaletteColor
+                ).dark,
               },
               '&.MuiButtonBase-root:active': {
                 // Changes back to the primary text color
                 color: `${theme.palette.text.primary} !important`,
                 // Changes background color back to main
-                backgroundColor: (theme.palette[ownerState.color] as PaletteColor).main,
+                backgroundColor: (
+                  theme.palette[ownerState.color as PaletteColorOption] as PaletteColor
+                ).main,
               },
             }
           : {
               '&.MuiButtonBase-root:hover': {
                 backgroundColor: `${
-                  (theme.palette[ownerState.color] as PaletteColor).light
+                  (theme.palette[ownerState.color as PaletteColorOption] as PaletteColor)
+                    .light
                 }`,
               },
               '&.MuiButtonBase-root:active': {
-                backgroundColor: (theme.palette[ownerState.color] as PaletteColor).main,
+                backgroundColor: (
+                  theme.palette[ownerState.color as PaletteColorOption] as PaletteColor
+                ).main,
               },
             })),
     }),
@@ -99,11 +105,13 @@ export default {
               }),
               // In dark mode, the text color lightens on hover
               '&.MuiButtonBase-root:hover': {
-                color: theme.palette[ownerState.color]['light'],
+                color: theme.palette[ownerState.color as PaletteColorOption]['light'],
               },
               // In dark mode, when in an active state, the text color dims back to normal
               '&.MuiButtonBase-root:active': {
-                color: `${theme.palette[ownerState.color]['main']} !important`,
+                color: `${
+                  theme.palette[ownerState.color as PaletteColorOption]['main']
+                } !important`,
               },
             })),
     }),

--- a/packages/ui-kit-theme/src/overrides/MuiDivider.ts
+++ b/packages/ui-kit-theme/src/overrides/MuiDivider.ts
@@ -17,11 +17,11 @@ export default {
         ownerState,
       }: {
         theme: Theme;
-        ownerState: { children: React.ReactChildren };
+        ownerState?: { children: React.ReactChildren };
       }) => {
         const styleProps = { borderColor: alpha(theme.palette.divider, 0.8) };
         // BorderColor overrides must be applied to (&::before, &::after) instead of the root when "children" prop is defined.
-        return ownerState.children
+        return ownerState && ownerState.children
           ? {
               '&::before, &::after': styleProps,
             }

--- a/packages/ui-kit-theme/src/overrides/MuiLinearProgress.ts
+++ b/packages/ui-kit-theme/src/overrides/MuiLinearProgress.ts
@@ -1,13 +1,13 @@
-import { Theme } from '@mui/material';
+import { LinearProgressProps, Theme } from '@mui/material';
 
 type OverrideContext = {
   theme: Theme;
-  ownerState: { value: number };
+  ownerState: LinearProgressProps;
 };
 
 export default {
   defaultProps: {
-    color: 'secondary',
+    color: 'secondary' as const,
   },
   styleOverrides: {
     root: {

